### ISSUE TITLE
[Bug]: Migration sqlite -> postgres value too long error

### ISSUE THREAD

**Issue by elvece** [btcpayserver/btcpayserver#4771]
### What is your BTCPay version?

Migrating from v1.7.7 -> 1.8.3 on an instance that was running sqlite.

### How did you deploy BTCPay Server?

embassyOS v0.3.3

### What happened?

The postgres migration failed to complete while updating to a new version.

### How did you encounter this bug?

1. Install btcpay v1.7.7 on embassyOS v0.3.3 (which is using a sqlite database)
2. Update btcpay to v1.8.3
3. Notice errors in logs while postgres data migration is being attempted

### Relevant log output

```shell
2023-03-14T10:10:10-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating from SQLite to Postgres...
2023-03-14T10:10:10-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: There is a unfinished migration in postgres... dropping all tables
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetRoles
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetRoleClaims
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUsers
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserClaims
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserLogins
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserRoles
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AspNetUserTokens
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Fido2Credentials
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Files
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Notifications
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: OffchainTransactions
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PairingCodes
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PayjoinLocks
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: PlannedTransactions
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Settings
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Stores
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: ApiKeys
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Apps
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: CustodianAccount
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Forms
2023-03-14T10:10:12-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: Invoices
2023-03-14T10:10:13-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: AddressInvoices
2023-03-14T10:10:13-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: InvoiceEvents
2023-03-14T10:10:13-06:00  info: BTCPayServer.Hosting.ToPostgresMigrationStartupTask: Migrating table: InvoiceSearches
2023-03-14T10:10:13-06:00  2023-03-14 16:10:13.113 UTC [1549] ERROR:  value too long for type character varying(512)
2023-03-14T10:10:13-06:00  2023-03-14 16:10:13.113 UTC [1549] STATEMENT:  INSERT INTO "InvoiceSearches" ("Id", "InvoiceDataId", "Value")
2023-03-14T10:10:13-06:00  	VALUES ($1, $2, $3)
2023-03-14T10:10:13-06:00  Unhandled exception. Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
2023-03-14T10:10:13-06:00   ---> Npgsql.PostgresException (0x80004005): 22001: value too long for type character varying(512)
2023-03-14T10:10:13-06:00     at Npgsql.Internal.NpgsqlConnector.<ReadMessage>g__ReadMessageLong|215_0(NpgsqlConnector connector, Boolean async, DataRowLoadingMode dataRowLoadingMode, Boolean readingNotifications, Boolean isReadingPrependedMessage)
2023-03-14T10:10:13-06:00     at Npgsql.NpgsqlDataReader.NextResult(Boolean async, Boolean isConsuming, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00    Exception data:
2023-03-14T10:10:13-06:00      Severity: ERROR
2023-03-14T10:10:13-06:00      SqlState: 22001
2023-03-14T10:10:13-06:00      MessageText: value too long for type character varying(512)
2023-03-14T10:10:13-06:00      File: varchar.c
2023-03-14T10:10:13-06:00      Line: 632
2023-03-14T10:10:13-06:00      Routine: varchar
2023-03-14T10:10:13-06:00     --- End of inner exception stack trace ---
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.NpgsqlExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
2023-03-14T10:10:13-06:00     at BTCPayServer.Hosting.ToPostgresMigrationStartupTask.ExecuteAsync(CancellationToken cancellationToken) in /source/BTCPayServer/Hosting/ToPostgresMigrationStartupTask.cs:line 258
2023-03-14T10:10:13-06:00     at Microsoft.AspNetCore.Hosting.WebHostExtensions.StartWithTasksAsync(IWebHost webHost, CancellationToken cancellationToken) in /source/BTCPayServer/Extensions/WebHostExtensions.cs:line 19
2023-03-14T10:10:13-06:00     at BTCPayServer.Program.Main(String[] args) in /source/BTCPayServer/Program.cs:line 76
```

### What browser do you use?

Firefox 110.0.1 

### Additional information

Instance has a basic setup with store invoicing and a crowdfund app.

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by NicolasDorier** [comment]
Quick fix by running this on the db:

```
ALTER TABLE "InvoiceSearches" ALTER COLUMN "Value" TYPE TEXT USING "Value"::TEXT;
ALTER TABLE "Invoices"  ALTER COLUMN "OrderId" TYPE TEXT USING "OrderId"::TEXT;
```

Making new release now fixing the issue without having to do manually.

**Comment by NicolasDorier** [comment]
fixed on 1.8.4
